db/commitlog/commitlog.cc

```diff
@@ -1584,7 +1583,8 @@ future<db::commitlog::segment_manager::sseg_ptr> db::commitlog::segment_manager:
                 // sync metadata (size/written)
                 co_await f.flush();
                 co_await f.close();
-                f = co_await open_file_dma(filename, flags, opt);
+                // filter out "create" so we re-read file size
+                co_await f.open(flags & open_flags(~int(open_flags::create)), opt);
```

Comment:
Again, I'm not sure if the right thing you wanted to filter was truncate, not create?

Reply:
In this case it is create actually, because "create" here always means new file (not recycled).

Reply:
But O_CREATE is not exclusive. It won't truncate or do anything to the file if it already exists.

Reply:
No, in that case we do manual truncation/fill to get to the right size (segment max size). Flipping the bit here is just to save some work for named_file::open. But we could really dispense with that too.

Reply:
Changed this to be more consistent. Caller tells size, or we truncate, or we do file::size().